src/Planners/src/UnicycleTrajectoryPlanner.cpp

```diff
@@ -0,0 +1,841 @@
+                                   comPlanarAcceleration.at(i)(1),
+                                   comHeightAcceleration.at(i)});
+    }
+    m_pImpl->output.comTrajectory.position = comPosition;
+    m_pImpl->output.comTrajectory.velocity = comVelocity;
+    m_pImpl->output.comTrajectory.acceleration = comAcceleration;
+
+    // get the merge points
+    std::vector<size_t> mergePoints;
+    m_pImpl->generator.getMergePoints(mergePoints);
+    m_pImpl->output.mergePoints = mergePoints;
+
+    m_pImpl->state = Impl::FSM::Running;
+
+    return true;
+}
+
+bool BipedalLocomotion::Planners::UnicycleTrajectoryPlanner::generateFirstTrajectory()
+{
+
+    constexpr auto logPrefix = "[UnicycleTrajectoryPlanner::generateFirstTrajectory]";
+
+    // clear the all trajectory
+    auto unicyclePlanner = m_pImpl->generator.unicyclePlanner();
+    unicyclePlanner->clearPersonFollowingDesiredTrajectory();
+    unicyclePlanner->setDesiredDirectControl(0.0, 0.0, 0.0);
+
+    // clear left and right footsteps
+    m_pImpl->generator.getLeftFootPrint()->clearSteps();
+    m_pImpl->generator.getRightFootPrint()->clearSteps();
+
+    // set initial and final times
+    double initTime = 0;
+    double endTime = initTime + m_pImpl->parameters.plannerHorizon;
+
+    // at the beginning ergoCub has to stop
+    Eigen::Vector2d m_personFollowingDesiredPoint;
+    m_personFollowingDesiredPoint(0) = m_pImpl->parameters.referencePointDistance(0);
+    m_personFollowingDesiredPoint(1) = m_pImpl->parameters.referencePointDistance(1);
+
+    // add the initial point
+    if (!unicyclePlanner
+             ->addPersonFollowingDesiredTrajectoryPoint(initTime,
+                                                        iDynTree::Vector2(
+                                                            m_personFollowingDesiredPoint)))
+    {
+        log()->error("{} Error while setting the initial point.", logPrefix);
+        return false;
+    }
+
+    // add the final point
+    if (!unicyclePlanner
+             ->addPersonFollowingDesiredTrajectoryPoint(endTime,
+                                                        iDynTree::Vector2(
+                                                            m_personFollowingDesiredPoint)))
+    {
+        log()->error("{} Error while setting the final point.", logPrefix);
+        return false;
+    }
+
+    // generate the first trajectories
+    if (!m_pImpl->generator.generate(initTime, m_pImpl->parameters.dt, endTime))
+    {
+
+        log()->error("{} Error while computing the first trajectories.", logPrefix);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool BipedalLocomotion::Planners::Utilities::getContactList(
+    const double initTime,
+    const double dt,
+    const std::vector<bool>& inContact,
+    const std::deque<Step>& steps,
+    const int contactFrameIndex,
+    const std::string& contactName,
+    BipedalLocomotion::Contacts::ContactList& contactList)
```

Comment:
It would be nice if Unicycle directly provided the contact list map or better a contact phase

Reply:
The [unicycle-footsteps-planner](https://github.com/robotology/unicycle-footstep-planner) is returning footsteps for example as `std::deque<Steps>`. Then I am converting the `deque` to a `ContactList` with `BipedalLocomotion::Planners::Utilities::getContactList`. Finally the `ContactList` of the two feet are used to populate the related `ContactListMap`, which are combined to form the `ContactPhaseList`, which is one of the output of the `UnicycleTrajectoryPlanner`.

A similar procedure is followed in the `UnicycleTrajectoryGenerator` of #845.

Reply:
see https://github.com/ami-iit/bipedal-locomotion-framework/pull/844#discussion_r1602805641